src/Application/Game.cpp

```diff
@@ -371,16 +371,13 @@ void Game_StartHirelingDialogue(unsigned int hireling_id) {
     uint hireling_slot = 0;
     char buf[1024] {};
     for (uint i = 0; i < 2; ++i) {
-        if (pParty->pHirelings[i].pName) buf[hireling_slot++] = i;
+        if (!pParty->pHirelings[i].pName.empty())
+            buf[hireling_slot++] = i;
     }
 
     for (uint i = 0; i < pNPCStats->uNumNewNPCs; ++i) {
         NPCData *npc = &pNPCStats->pNewNPCData[i];
-        if (npc->Hired() &&
-            (!pParty->pHirelings[0].pName ||
-             strcmp(npc->pName, pParty->pHirelings[0].pName)) &&
-            (!pParty->pHirelings[1].pName ||
-             strcmp(npc->pName, pParty->pHirelings[1].pName))) {
+        if (npc->Hired() && npc->pName != pParty->pHirelings[0].pName && npc->pName != pParty->pHirelings[1].pName) {
```

Comment:
Is pParty->pHirelings[x].pName.empty check really not neccessary here (and all other similary simplified places) in all cases?

Reply:
Not necessary unless we have nameless NPCs. You think this is possible?

Reply:
It will work unless we actually have NPCs with empty names. Shouldn't happen, but I can add assertions to check for that.

Reply:
Went through the changes again, these checks are always preceded by a check for Hired(), and I believe that if NPC is Hired then it should have a name. I'll go through this code again in a separate PR.

Reply:
OK just moved this code out into a separate class.

Reply:
OK just moved this code out.